src/main/java/com/cms/users/entity/UserBase.java

```diff
@@ -0,0 +1,58 @@
+package com.cms.users.entity;
+
+import java.security.SecureRandom;
+
+import javax.crypto.Cipher;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import javax.persistence.PrePersist;
+
+import org.apache.commons.codec.binary.Base64;
+import org.slf4j.LoggerFactory;
+
+public class UserBase {
+	public static org.slf4j.Logger log = LoggerFactory.getLogger(Participation.class);
+	
+	
+	/*
+	 *This method creates a new encrypted password
+	 *Using as main key the string in the cms-config file
+	 * 
+	 * */
+	@PrePersist
+	public void userPrePersist(User user) {
+		String newPassword = "";
+		
+		try {
+		String input = user.getPassword();
+		String key = "8e045a51e4b102ea803c06f92841a1fb";
+		String initVector = "RandomInitVector";//Random Key
+		
+		 IvParameterSpec iv = new IvParameterSpec(initVector.getBytes("UTF-8"));
+		 SecretKeySpec skeySpec = new SecretKeySpec(key.getBytes("UTF-8"), "AES");
+		
+		 Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5PADDING");
```

Comment:
I'm curious, are we storing the CMS users password as plaintext or as bcript value?

Reply:
Right now I'm working with plain/text but it needs to be AES iv